staging/src/k8s.io/dynamic-resource-allocation/structured/allocator.go

```diff
@@ -972,6 +972,15 @@ func (alloc *allocator) allocateDevice(r deviceIndices, device deviceWithID, mus
 		return false, nil, nil
 	}
 
+	// Since adminAccess allocations are not tracked, check again that the
+	// device hasn't already been allocated for this claim.
+	if request.adminAccess() && slices.ContainsFunc(alloc.result[r.claimIndex].devices, func(previousResult internalDeviceResult) bool {
```

Comment:
This check is probably valid to do for non-admin access claims as well, but redundant in the majority of cases. Only doing this for admin access claims allows this condition to short-circuit before searching the list of allocated devices linearly.